Services/Store/ModsenOnlineStore.Store.Infrastructure/Data/OrderRepository.cs

```diff
@@ -13,9 +13,21 @@ public OrderRepository(DataContext context)
             this.context = context;
         }
 
-        public async Task<List<Order>> GetAllOrders()
+        public async Task<List<Order>> GetAllOrders(int pageNumber, int pageSize)
         {
-            return await context.Orders.AsNoTracking().ToListAsync();
+            var orders = await context.Orders.AsNoTracking().ToListAsync();
+
+            if (pageNumber < 1)
+            {
+                return orders;
+            }
+
+            if (pageSize < 1)
+            {
+                pageSize = 10;
+            }
```

Comment:
I see that this code is repeating over and over again
May be there is a solution to write it once and use it everywhere needed

Reply:
I created a generic abstract class with the protected method to which i moved this code. All repositories are inherited from this class.